_posts/2020-05-25-pinot-connector.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title:  "Apache Pinot Connector"
+author: Elon Azoulay
+---
+
+#In an upcoming release of Presto
+Introducing the new Apache Pinot™ Connector which allows Presto to query data stored in 
+[Apache Pinot™ Connector](https://pinot.apache.org/). 
+Not only does this allow access to pinot tables but gives users the ability to do things they could not do with pinot 
+alone such as join Pinot tables to other tables and use Presto's UDF's, window functions and complex aggregations.
+
+Pinot udf's can be directly used by including the pinot sql query in quotes, explained below in the "Dynamic Sql" section. 
+This enables aggregations and other complex query types to be done directly in pinot.
+
+This connector supports pinot-0.3.0 and newer.
+
+#Setup
+Create a properties file in the catalog directory, such as `etc/catalog/pinot.properties` which includes atleast the
+following to get started:
+
+```
+connector.name=pinot
+pinot.controller-urls=host1:9000,host2:9000
+```
+
+The `pinot.controller-urls` is a comma separated list of controller hosts. If Pinot is deployed via [Kubernetes](https://kubernetes.io/) and you expose the 
+the `pinot.controller-urls` needs to point to the controller Service endpoint. The Pinot broker and server must be accessible
+via dns as Pinot will return hostnames and not ip addresses.
+
+If you have a smaller number of Pinot servers than Presto workers or a relatively small number of rows per Pinot segment,
+you can minimize the requests to pinot by increasing the number of Pinot segments per split (default is 1 segment per split):
+
+```
+pinot.segments-per-split=15
+```
+
+If dns resolution is slow or you get `Request timed out` errors, you can increase the request timeout as follows:
+
+```
+pinot.request-timeout=3m
+```
+ 
+#Schema
+Pinot supports the following data types. Currently null values are not supported. The corresponding Presto datatypes are:
+
+| Pinot Datatype | Presto Datatype |
+| -------------- | --------------- |
+| boolean | boolean |
+| integer | integer |
+| float, double | double |
+| string, bytes* | varchar |
+| integer_array | array<integer> |
+| float_array, double_array | array<double> |
+| string_array | array<varchar> |
+
+* The pinot `bytes` type is converted to a hex encoded varchar. See the [pinot docs](https://pinot.apache.org/) for more information.  
+
+#Dynamic Sql
+If you would like to leverage Pinot's fast aggregations you can use a "dynamic" table where you specify the Calcite SQL 
```

Comment:
Calcite is an implementation detail, I believe. I'd say Pinot SQL instead.

Reply:
is Calcite SQL the same as Pinot PQL? If so, the let's use PQL everywhere we talk about raw pinot sql or dynamic sql

Reply:
It's different: PQL is being deprecated and removed soon. Will just use Pinot SQL - since calcite is the parser (implementation detail) and not all constructs are implemented yet (i.e. joins, struct types, case statements, etc.)